app/src/main/java/ca/llamabagel/transpo/search/ui/SearchViewModel.kt

```diff
@@ -49,10 +54,22 @@ class SearchViewModel @Inject constructor(
     }
 
     fun fetchSearchResults(query: CharSequence?) {
-        val queryString = query?.toString().orEmpty()
+        searchQuery = query?.toString().orEmpty()
+
+        viewModelScope.launch {
+            updateQuery(searchQuery, searchFilters)
+        }
+    }
+
+    fun notifyFilterChanged(@IdRes item: Int) {
+        when (item) {
+            R.id.routes_filter -> searchFilters.apply { routes = !routes }
```

Comment:
```suggestion
            R.id.routes_filter -> searchFilter = searchFilter.copy(routes = !searchFilter.routes)
```

Reply:
oh god i'm gonna have to right so many more tests